modules/ipcCommunicator.js

```diff
@@ -32,32 +34,38 @@ ipc.on('backendAction_closeApp', function() {
     app.quit();
 });
 ipc.on('backendAction_closePopupWindow', function(e) {
-    var windowId = e.sender.getId();
+    var windowId = e.sender.getId(),
+        senderWindow = global.windows[windowId];
 
-    if(global.windows[windowId]) {
-        global.windows[windowId].window.close();
+    if(senderWindow) {
+        senderWindow.window.close();
         delete global.windows[windowId];
     }
 });
 ipc.on('backendAction_setWindowSize', function(e, width, height) {
-    var windowId = e.sender.getId();
+    var windowId = e.sender.getId(),
+        senderWindow = global.windows[windowId];
 
-    if(global.windows[windowId]) {
-        global.windows[windowId].window.setSize(width, height);
-        global.windows[windowId].window.center(); // ?
+    if(senderWindow) {
+        senderWindow.window.setSize(width, height);
+        senderWindow.window.center(); // ?
     }
 });
 
 ipc.on('backendAction_sendToOwner', function(e, error, value) {
-    var windowId = e.sender.getId();
+    var windowId = e.sender.getId(),
+        senderWindow = global.windows[windowId];
 
-    if(global.windows[windowId]) {
-        global.windows[windowId].owner.send('windowMessage', global.windows[windowId].type, error, value);
+    var mainWindow = global.mainWindow;
 
+    if (_.get(senderWindow, 'owner')) {
```

Comment:
Are you're `_.get()` exists in underscore?
Im fine using lodash if you think its a better fit, as long as it also supports the api of underscore